sdk/eidolon_ai_sdk/agent/retriever_agent/retriever.py

```diff
@@ -1,28 +1,30 @@
 import asyncio
-from typing import List
+from typing import AsyncIterable
 
 from pydantic import BaseModel, Field
 
 from eidolon_ai_sdk.agent.retriever_agent.document_reranker import DocumentReranker
+from eidolon_ai_sdk.agent.retriever_agent.document_retriever import DocumentRetriever
 from eidolon_ai_sdk.agent.retriever_agent.question_transformer import QuestionTransformer
+from eidolon_ai_sdk.agent.retriever_agent.result_summarizer import ResultSummarizer, DocSummary
 from eidolon_ai_sdk.agent_os import AgentOS
 from eidolon_ai_sdk.system.reference_model import Specable, AnnotatedReference
 
 
 class RetrieverSpec(BaseModel):
-    max_num_results: int = Field(default=10, description="The maximum number of results to send to cpu.")
+    max_num_results: int = Field(default=10, description="The maximum number of results to consider.")
     question_transformer: AnnotatedReference[QuestionTransformer]
+    document_retriever: AnnotatedReference[DocumentRetriever]
     document_reranker: AnnotatedReference[DocumentReranker]
-
-
-class DocSummary(BaseModel):
-    id: str
-    file_name: str
-    file_path: str
-    text: str
+    result_summarizer: AnnotatedReference[ResultSummarizer]
 
 
 class Retriever(Specable[RetrieverSpec]):
+    document_retriever: DocumentRetriever
+    document_reranker: DocumentReranker
+    question_transformer: QuestionTransformer
```

Comment:
could you change the type annotation here to be Optional[QuestionTransformer]

Reply:
You think we should do that or make it explicit? You likely never want the basic one that just uses the query against the DB